File: frontend/src/types/persona.ts
// 用户画像相关类型 — 与后端 PersonaVO / PersonaTaskStatusVO 字段保持一致

export interface PersonaPainPoint {
  title: string
  description: string
}

export type PaymentLevel = 'high' | 'medium' | 'low'

export interface PersonaDecisionParams {
  paymentWillingness: PaymentLevel
  decisionCycle: string   // "1-2 周" / "2-4 周" / "2-6 月"
  budgetRange: string     // "0.5-2 万/年" / "3-8 万/年" / "10-50 万/年"
  techCapability: PaymentLevel
}

export interface Persona {
  id: number
  projectId: number
  name: string
  role: string
  ageRange: string
  marketShare: number
  isPrimary: boolean
  goals: string[]
  painPoints: PersonaPainPoint[]
  quote: string
  decisionParams: PersonaDecisionParams
  createdAt: string
  updatedAt: string
}

export interface GeneratePersonaRequest {
  projectId: number
  count: number       // 3-5
  context?: string
}

export interface CreatePersonaTaskResponse {
  taskId: number
  status: string
  estimatedSeconds: number
}

export type PersonaTaskStatus = 'pending' | 'running' | 'completed' | 'failed'

export interface PersonaTaskInfo {
  taskId: number
  status: PersonaTaskStatus
  progress: number
  projectId: number
  errorMessage?: string
}

// 手动创建画像
export interface CreatePersonaPayload {
  projectId: number
  name: string
  role: string
  ageRange: string
  marketShare?: number
  isPrimary?: boolean
  goals?: string[]
  painPoints?: PersonaPainPoint[]
  quote?: string
  decisionParams?: PersonaDecisionParams
}

// 更新画像 (所有字段可选)
export type UpdatePersonaPayload = Partial<Omit<CreatePersonaPayload, 'projectId'>>

File: backend/src/main/java/com/marketai/backend/entity/Persona.java
package com.marketai.backend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 用户画像实体
 * JSON 列 (goals/painPoints/decisionParams) 与 InsightReport 同样以原始字符串存储,
 * 由 Service 层使用 ObjectMapper 解析。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("personas")
public class Persona {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;

    private String name;
    private String role;
    private String ageRange;

    /** 在目标市场中的占比 (百分比, 0-100) */
    private BigDecimal marketShare;

    /** 0=普通 / 1=核心用户 */
    private Integer isPrimary;

    private String goals;          // JSON: ["目标1", "目标2"]
    private String painPoints;     // JSON: [{title, description}]
    private String quote;
    private String decisionParams; // JSON: {paymentWillingness, decisionCycle, budgetRange, techCapability}

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
